imagebuildah/stage_executor.go

```diff
@@ -1948,7 +1969,7 @@ func (s *StageExecutor) commit(ctx context.Context, createdBy string, emptyLayer
 		Manifest:              s.executor.manifest,
 	}
 	// generate build output
-	if s.executor.buildOutput != "" {
+	if s.executor.buildOutput != "" && lastStage {
 		if err := s.generateBuildOutput(buildah.CommitOptions{}, buildOutputOption); err != nil {
```

Comment:
Does `generateBuildOutput()` benefit from any of the things that `commit()` is doing before this point?  If not, why not just have the rest of the places that call `commit()` call `generateBuildOutput()` directly when they need to, instead of making it a side-effect?

Reply:
Fixed